lib/iris/_deprecation_helpers.py

```diff
@@ -24,10 +24,20 @@
 
 import warnings
 
+from iris.exceptions import IrisError
+
+
+class IrisDeprecation(IrisError, DeprecationWarning):
```

Comment:
> one for deprecations and one for changes in behaviour?

Whatever the reason I did this, that wasn't it !

I don't think I had a good reason for this, I might have just been thinking that IrisDeprecation  should belong in iris.exceptions.
Probably best just removed.